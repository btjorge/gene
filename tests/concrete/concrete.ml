open Genelib

let prefix = "../progs/"

let progs = [ (*"max01";*) "max02"; "max03"; "max04"; "max05"; "max07";
              "max08"; "max09"; "max10"; "max11"; "max12"; "max3"; "max_imp";
              "max_imp2" ]

let interpret file =
  let ic = open_in (prefix ^ file) in
  let lexbuf = Lexing.from_channel ic in

  let p = Parser.prog Lexer.token lexbuf in

  let _ = Interpreter.eval_prog p in
  ()

let () =
  List.iter interpret progs
