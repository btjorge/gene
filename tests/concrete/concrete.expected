assertion passed: max (5, 2) = 5
assertion passed: max (5, 2) = 5
assertion failed: 2 != 5
assertion failed: 2 != 5
assertion passed: max (5, 2) = 5
assertion failed: 5 != 6
assertion passed: max (5, 2) = 5
assertion passed: max (6, 2) = 6
assertion passed: max (5, 2) = 5
assertion passed: max (6, 2) = 6
assertion passed: max (2, 5) = 5
assertion passed: max (5, 2) = 5
assertion passed: max (6, 2) = 6
assertion passed: max (2, 5) = 5
assertion failed: 2 != 5
assertion failed: 2 != 6
assertion failed: 2 != 5
assertion passed: max (5, 2) = 5
assertion passed: max (6, 2) = 6
assertion passed: max (2, 5) = 5
assertion passed: max3 (1, 3, 2) = 3
assertion passed: z = 3
assertion passed: z = y
